packages/node-integration-tests/src/common/mock-segment.ts

```diff
@@ -0,0 +1,78 @@
+import { createServer, IncomingMessage } from 'http'
+
+async function getRequestText(req: IncomingMessage) {
+  return new Promise<string>((resolve, reject) => {
+    const data: Buffer[] = []
+    req.on('error', reject)
+    req.on('data', (chunk) => {
+      data.push(chunk)
+    })
+    req.on('end', () => {
+      resolve(Buffer.concat(data).toString())
+    })
+  })
+}
+
+function isBatchRequest(req: IncomingMessage) {
+  if (req.url?.endsWith('/v1/batch')) {
+    return true
+  }
+  return false
+}
+
+type BatchHandler = (batch: any) => void
+
+export class MockSegmentServer {
```

Comment:
You know we have this, which also is a mock segment server?  https://github.com/segmentio/analytics-next/tree/master/packages/node-integration-tests/src/server